### PR TITLE
WIP Increase print efficiency by reducing travel distance

### PR DESCRIPTION
**Pull request by tsdobbs**

### CODE REVIEW

**Comment by tsdobbs** [comment]
@patrickjyoung Is the Point class used at all here?

File: web/robot/list_reordering.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            min_point = v
+
+    return min_point
+
+def euclideanDistance(start, end):
+    return math.sqrt((start[0] - end[0])**2 +(start[1] - end[1])**2)
+
+
+
+    # Funtion that implements Dijkstra's single source
```
Comment (tsdobbs) [inline web/robot/list_reordering.py:34]: Are these comments accurate for this version of the code?

File: web/robot/list_reordering.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import sys
+import math
+import random
+
+class Point():
```
Comment (tsdobbs) [inline web/robot/list_reordering.py:5]: Is this class used?

File: web/robot/list_reordering.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        self.x = x;
+        self.y = y;
+
+    def __eq__(self, other):
+        return (self.x == other.x) & (self.y == other.y)
+
+
+def minDistance(start, remaininglist):
+
+        # Initilaize minimum distance for next node
```
Comment (tsdobbs) [inline web/robot/list_reordering.py:16]: Align comments with tabbing over the code they comment

File: web/robot/list_reordering.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+class Point():
+    def __init__(self, x, y):
+        self.x = x;
+        self.y = y;
+
+    def __eq__(self, other):
+        return (self.x == other.x) & (self.y == other.y)
+
+
```
Comment (tsdobbs) [inline web/robot/list_reordering.py:13]: A more verbose function name might be helpful here. Something like "find_closest_neighbor". Also note the function naming style for this repo is _lowercase_with_underscores_

File: web/robot/list_reordering.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        # shortest path tree
+    for v in remaininglist:
+        dist = euclideanDistance(start, v)
+        if dist < min:
+            min = dist
+            min_point = v
+
+    return min_point
+
+def euclideanDistance(start, end):
```
Comment (tsdobbs) [inline web/robot/list_reordering.py:29]: _euclidean_distance_ matches the function naming convention for BioArtBot

**Review by tsdobbs — COMMENTED** [review]
This is a great start. The algorithm itself can certainly be made more efficient, but I'm not sure if it's worth it, given our relatively small number of possible points. Before we merge, can you do a review of solutions to problems that are like [The Travelling Salesman Problem](https://en.wikipedia.org/wiki/Travelling_salesman_problem)? From there we can decide if a different algorithm would be worthwhile.

Additionally, if you can write a test with known inputs/outputs, I can show you how to add it to our test pipeline so it gets tested automatically when deployed.

File: web/robot/list_reordering.py (hunk 1 above)
Comment (patrickjyoung) [reply web/robot/list_reordering.py:34]: No, had left them in, will update soon.

File: web/robot/list_reordering.py (hunk 2 above)
Comment (patrickjyoung) [reply web/robot/list_reordering.py:5]: Not with the version I sent you, originally I had assumed that the code had objects as Points with x and y properties.

**Review by patrickjyoung — APPROVED** [review]

**Pull request merged**